### PR TITLE
Adding default_value and hidden properties to custom privacy request fields

### PR DESCRIPTION
**Pull request by galvana**
### Description Of Changes

Adding the ability to support hidden custom privacy request fields. This requires the addition of a default value so I added two new properties `default_value` and `hidden`.

- The default value for a hidden field won't be shown in the form, but it will be included in the privacy request payload
- The default value for a non-hidden field will be shown in the form and can be overwritten by the user

I also limited the height of the privacy request form field and added scrolling.

### Code Changes

* [ ] Updated `smoke_test.cy.ts`
* [ ] Added validation to error fields flagged as hidden without a default value
* [ ] Updated styling on the form
* [ ] Updated form logic to account for the new field properties

### Steps to Confirm

* [x] Enable custom privacy request field collection https://docs.ethyca.com/dev-docs/configuration/privacy-requests/custom-request-fields#enable-custom-fields
* [x] Run `nox -s "fides_env(test)"`
* [x] Navigate to the Privacy Center
* [x] Fill out the form for an access request
* [x] Navigate to the Admin UI
* [ ] Verify the Tenant ID is included even though it wasn't shown to the user in the access request form

### Pre-Merge Checklist

* [ ] All CI Pipelines Succeeded
* Documentation:
  * [ ] documentation complete, [PR opened in fidesdocs](https://github.com/ethyca/fidesdocs/pulls)
  * [ ] documentation [issue created in fidesdocs](https://github.com/ethyca/fidesdocs/issues/new/choose)
* [ ] Issue Requirements are Met
* [ ] Relevant Follow-Up Issues Created
* [ ] Update `CHANGELOG.md`
* [ ] For API changes, the [Postman collection](https://github.com/ethyca/fides/blob/main/docs/fides/docs/development/postman/Fides.postman_collection.json) has been updated

### CODE REVIEW

File: clients/cypress-e2e/cypress/e2e/smoke_test.cy.ts [hunk 1]
```diff
@@ -12,16 +12,49 @@ describe("Smoke test", () => {
...
+              value: "",
+            },
+            color: {
+              label: "Color",
+              value: "blue",
+            },
+            tenant_id: {
+              label: "Tenant ID",
+              value: "123",
+            },
```
Comment (galvana) [inline clients/cypress-e2e/cypress/e2e/smoke_test.cy.ts:51]: Verifying a few scenarios with this check

- A non-required field still contains a value, even if it's an empty string
- A field with a `default_value` can be overwritten
- A hidden field with a `default_value` is still included with the other form values in the request

Comment (galvana) [inline clients/cypress-e2e/cypress/e2e/smoke_test.cy.ts:51]: Verifying a few scenarios with this check

- A non-required field still contains a value, even if it's an empty string
- A field with a `default_value` can be overwritten in the form
- A hidden field with a `default_value` is still included with the entered form values in the request

File: clients/privacy-center/app/server-environment.ts [hunk 2]
```diff
@@ -178,6 +178,21 @@ export const validateConfig = (
...
   }
+
+  // Validate that each hidden field in custom_privacy_request_fields has a default value
+  config.actions.forEach((action) => {
+    Object.entries(action.custom_privacy_request_fields || {}).forEach(
+      ([key, field]) => {
+        if (field.hidden && field.default_value === undefined) {
+          return {
+            isValid: false,
+            message: `Default value required for hidden field '${key}' in action '${action.policy_key}'`,
```
Comment (galvana) [inline clients/privacy-center/app/server-environment.ts:189]: This error message doesn't surface completely, but I see that's a known issue https://github.com/ethyca/fides/issues/3171

File: clients/privacy-center/components/modals/privacy-request-modal/PrivacyRequestForm.tsx [hunk 3]
```diff
@@ -273,11 +273,11 @@ const PrivacyRequestForm: React.FC<PrivacyRequestFormProps> = ({
...
       </ModalHeader>
+      <Text fontSize="sm" color="gray.600" mb={4} ml={6}>
+        {action.description}
+      </Text>
       <chakra.form onSubmit={handleSubmit} data-testid="privacy-request-form">
-        <ModalBody>
-          <Text fontSize="sm" color="gray.600" mb={4}>
-            {action.description}
-          </Text>
+        <ModalBody maxHeight={400} overflowY="auto">
```
Comment (galvana) [inline clients/privacy-center/components/modals/privacy-request-modal/PrivacyRequestForm.tsx:280]: Updating the popup to scroll if the form body exceeds 400px

File: clients/privacy-center/app/server-environment.ts (hunk 2 above)
Comment (allisonking) [reply clients/privacy-center/app/server-environment.ts:189]: this doesn't appear to be working—I set a value to `hidden` without a `default_value` but it still says `isValid = true`! 

I think `forEach` `return`s don't work quite as we expect them to here: https://stackoverflow.com/questions/56129153/how-to-return-a-value-from-foreach

**Review by allisonking — CHANGES_REQUESTED** [review]
confirmed happy path works! 
image

I don't think the validation is working quite right though

**Review by allisonking — APPROVED** [review]
🙌

**Pull request merged**